src/tcp_mbedtls.c

```diff
@@ -325,6 +325,11 @@ int tcp_ssl_new_psk_client(struct tcp_pcb *tcp, void *arg, const char* psk_ident
 
   int ret = 0;
 
+  if (pskey == NULL || psk_ident == NULL) {
+    TCP_SSL_DEBUG(" failed\n  !  pre-shared key or identity is NULL\n\n");
+    return -1;
+  }
```

Comment:
This should go at the top of the function, before stuff is allocated, else you have to clean up properly.

Reply:
you're right